src/JustSaying/AwsTools/MessageHandling/QueueWithAsyncStartup.cs

```diff
@@ -6,27 +6,26 @@ namespace JustSaying.AwsTools.MessageHandling
     /// Represents an <see cref="ISqsQueue"/> with an associated startup task that must be run before
     /// the queue is ready to be used.
     /// </summary>
-    /// <typeparam name="TQueue">The type of the SQS queue.</typeparam>
-    public class QueueWithAsyncStartup<TQueue> where TQueue : ISqsQueue
+    public class QueueWithAsyncStartup
     {
         /// <summary>
-        /// Creates an instance of <see cref="QueueWithAsyncStartup{TQueue}"/> for which the <see cref="TQueue"/>
+        /// Creates an instance of <see cref="QueueWithAsyncStartup"/> for which the <see cref="Queue"/>
```

Comment:
Should it be `ISqsQueue`?

```suggestion
        /// Creates an instance of <see cref="QueueWithAsyncStartup"/> for which the <see cref="ISqsQueue"/>
```

Reply:
I thought that referring to the  instance was more specific, however I could go either way

Reply:
No, you're right, this is the constructor so the call site is concerned with what to pass in.